jwst/outlier_detection/outlier_detection.py

```diff
@@ -386,11 +386,14 @@ def flag_cr(sci_image, blot_image, **pars):
     snr1, snr2 = [float(val) for val in pars.get('snr', '5.0 4.0').split()]
     scl1, scl2 = [float(val) for val in pars.get('scale', '1.2 0.7').split()]
 
-    if not sci_image.meta.background.subtracted:
-        # Include background back into blotted image for comparison
+    # Include subtracted background back into blotted image for comparison
+    if (sci_image.meta.background.subtracted is False and
```

Comment:
I'm confused. If ``background.subtracted`` is False that would imply that no background has been subtracted from this image (i.e. the background signal is still in the image), so there shouldn't be any need to add it back in (it's already there).

Reply:
Agree.  I suspect the logic is incorrect.  Fixing.

Maybe @stsci-hack has thoughts on this?

Are we squashing two bugs for the price of one here?

Reply:
@hbushouse is right, and the changes look like they are more logical than what was there before.  Unfortunately, I can't remember why it was implemented that way in the first place...